### PR TITLE
Makefile cleanup

### LINKED ISSUE #1146: Clean up and document makefile
The makefile could do with a little documentation, and there may be some obsolete targets we can clean up.

### PR DESCRIPTION
**Pull request by codysoyland**
## Overview

Remove obsolete code. Turn off the automatic installation of build-time dependencies, which was confusing and inconsistent. Update PHONY list. Simplify git status check. Remove PKGS as Go 1.9+ automatically ignores `vendor` when running tests. Consolidate release/prerelease build code. Add comments. Add documentation.

Fixes #1146 

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,123 +1,133 @@
...
-
-dep: $(GOPATH)/bin
-	go get -u github.com/golang/dep/cmd/dep
-
+# Set up vendor directory using `dep`
 vendor: Gopkg.toml
-ifndef DEP
-	make dep
-endif
+	$(MAKE) require-dep
```
Comment (jaffee) [inline Makefile]: I don't understand how $(MAKE) works - where is that set?

Comment (codysoyland) [reply Makefile]: The `make` manual advises to always use `$(MAKE)`, which subtly changes a few things. Details are here: https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html

File: Makefile [hunk 2]
```diff
@@ -1,123 +1,133 @@
...
+install-dep:
+	go get -u github.com/golang/dep/cmd/dep
+
+install-statik:
+	go get -u github.com/rakyll/statik
+
+install-protoc-gen-gofast:
+	go get -u github.com/gogo/protobuf/protoc-gen-gofast
+
+install-protoc:
```
Comment (alanbernstein) [inline Makefile]: https://google.github.io/proto-lens/installing-protoc.html has some slightly more detailed installation instructions

File: Makefile [hunk 3]
```diff
@@ -1,123 +1,133 @@
...
-Gopkg.lock: dep Gopkg.toml
-	dep ensure
-
+# Run test suite
 test: vendor
-	go test $(PKGS) $(TESTFLAGS)
+	go test ./... $(TESTFLAGS)
 
+# Run test suite with coverage enabled
 cover: vendor
```
Comment (alanbernstein) [inline Makefile]: I think this is assuming the build directory exists

File: Makefile [hunk 4]
```diff
@@ -1,123 +1,134 @@
...
-release:
-ifeq ($(STATUS),"")
-	make release-build GOOS=darwin GOARCH=amd64
-	make release-build GOOS=linux GOARCH=amd64 DOCKER_BUILD=1
-	make release-build GOOS=linux GOARCH=386 DOCKER_BUILD=1
-else
-	@echo "Will not create release with unclean git status."
-endif
-
+	$(MAKE) $(if DOCKER_BUILD,docker-)build FLAGS="-o build/pilosa-$(VERSION_ID)/pilosa"
```
Comment (alanbernstein) [inline Makefile]: looks like there should also be a `mkdir` in here and `release-build`

Comment (codysoyland) [reply Makefile]: I've tested that the go build process will create the directory automatically, so I think it's only needed in that one place.

**Pull request merged**